spec/lib/cds_importer/entity_mapper/measure_mapper_spec.rb

```diff
@@ -56,4 +56,187 @@
+          'partialTemporaryStopRegulationId' => 'R1312020',
+          'partialTemporaryStopRegulationOfficialjournalNumber' => 'L 321',
+          'partialTemporaryStopRegulationOfficialjournalPage' => '1',
+          'abrogationRegulationId' => 'R1312021',
+          'abrogationRegulationOfficialjournalNumber' => 'L 323',
+          'abrogationRegulationOfficialjournalPage' => '2',
+          'metainfo' => {
+            'opType' => operation,
+            'transactionDate' => '2017-07-25T21:03:21',
+          },
+        },
+      }
+    end
+
+    before do
+      create(
+        :measure,
+        :with_footnote_association,
+        :with_measure_components,
+        :with_measure_conditions,
+        :with_measure_excluded_geographical_area,
+        :with_measure_partial_temporary_stop,
+        measure_sid: '12348',
+      )
+    end
+
+    context 'when the measure is being updated' do
+      let(:operation) { 'U' }
+
+      it_behaves_like 'an entity mapper update operation', Measure
+      it_behaves_like 'an entity mapper update operation', FootnoteAssociationMeasure
+      it_behaves_like 'an entity mapper update operation', MeasureComponent
+      it_behaves_like 'an entity mapper update operation', MeasureCondition
+      it_behaves_like 'an entity mapper update operation', MeasureConditionComponent
+      it_behaves_like 'an entity mapper update operation', MeasureExcludedGeographicalArea
+      it_behaves_like 'an entity mapper update operation', MeasurePartialTemporaryStop
+    end
+
+    context 'when the measure is being created' do
+      let(:operation) { 'C' }
+
+      it_behaves_like 'an entity mapper create operation', Measure
+      it_behaves_like 'an entity mapper create operation', FootnoteAssociationMeasure
+      it_behaves_like 'an entity mapper create operation', MeasureComponent
+      it_behaves_like 'an entity mapper create operation', MeasureCondition
+      it_behaves_like 'an entity mapper create operation', MeasureConditionComponent
+      it_behaves_like 'an entity mapper create operation', MeasureExcludedGeographicalArea
+      it_behaves_like 'an entity mapper create operation', MeasurePartialTemporaryStop
+    end
+
+    context 'when the measure is being deleted' do
+      let(:operation) { 'D' }
+
+      it_behaves_like 'an entity mapper destroy operation', Measure
+      it_behaves_like 'an entity mapper destroy operation', FootnoteAssociationMeasure
+      it_behaves_like 'an entity mapper destroy operation', MeasureComponent
+      it_behaves_like 'an entity mapper destroy operation', MeasureCondition
+      it_behaves_like 'an entity mapper destroy operation', MeasureConditionComponent
+      it_behaves_like 'an entity mapper destroy operation', MeasureExcludedGeographicalArea
+      it_behaves_like 'an entity mapper destroy operation', MeasurePartialTemporaryStop
+    end
+
+    context 'when there are missing secondary entities to be soft deleted' do
```

Comment:
I don't understand this test (and the other with "... to be soft deleted" ,
What is the feature in this test that makes me think that the missing secondary entities have to be soft deleted?

Reply:
Added some context that should help :smile: